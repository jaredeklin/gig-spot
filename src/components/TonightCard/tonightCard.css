.tonight-card {
  display: inline-block;
  width: 250px;
  height: 250px;
  margin: 10px;
  /*cursor: pointer;*/
  /*transition: 450ms all;*/
  /*transform-origin: center left;*/
  white-space: initial;
  /*vertical-align: bottom;*/
  box-sizing: border-box;
  opacity: .8;
  position: relative;
}

.tonight-card:hover {
  opacity: 1;
}

.card-info {
  position: absolute;
  bottom: 0;
  color: white;
  background-color: rgba(0 ,0 , 0, .5);
  width: 100%;
  padding: 5px 5px;
  box-sizing: border-box;
}

.artist {
  font-size: 16px
}

.venue {
  margin: 2px 0;
  font-size: 14px
}

.time {
  display: none;
}

.date {
  font-size: 14px;
}

.tonights-shows .tonight-card {
  width: 300px;
  height: 300px;
}

.tonights-shows .card-info {
  /*width: 300px;*/
  padding: 5px 10px;
}

.tonights-shows .artist {
  font-size: 20px;
}

.tonights-shows .venue {
  font-size: 16px;
}

.tonights-shows .date {
  display: none;
}

.tonights-shows img {
  width: 300px;
}

.tonights-shows .time {
  display: unset;
  font-size: 16px;
}


