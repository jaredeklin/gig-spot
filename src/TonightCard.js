import React from 'react';
import './tonightCard.css'

export const TonightCard = (props) => {
  const { image, artist, venue, date } = props.show;
  console.log(props)
  return (
    <article className="tonight-card">
      <img src={image} className='artist-image'/>
      <h4 className="artist">{artist}</h4>
      <h4 className="venue">{venue}</h4>
    </article>
  )
}