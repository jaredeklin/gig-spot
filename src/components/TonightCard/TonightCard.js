import React from 'react';
import './tonightCard.css';
import defaultImage from '../../images/black-woven.jpg';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export const TonightCard = ({ show }) => {

  let { image, headlineArtist, venue, date, startTime, id } = show;
  
  return (
    <Link to={`event-details/${id}`}>
      <article className="tonight-card">
        <img 
          src={image} 
          onError={(event) => event.target.src = defaultImage}
          className="artist-image" 
          alt="artist"
        />
        <div className="card-info">
          <h4 className="artist">{ headlineArtist }</h4>
          <div className="venue">{ venue.name }</div>
          <div className="date">{ date }</div>
          <div className="time">{ startTime }</div>
        </div>
      </article>
    </Link>
  );
};

TonightCard.propTypes = {
  id: PropTypes.object,
  show: PropTypes.object
};