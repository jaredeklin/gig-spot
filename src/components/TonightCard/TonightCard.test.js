import React from 'react';
import { shallow } from 'enzyme';
import { TonightCard } from './TonightCard';
import { 
  mockTonightCardData, 
  mockTonightCardDataWOImage 
} from '../../cleaners/mockData.js';

describe('TonightCard', () => {

  it('should match the snapshot', () => {
    let wrapper = shallow(<TonightCard show={mockTonightCardData} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot', () => {
    let wrapper = shallow(<TonightCard show={mockTonightCardDataWOImage} />);
    expect(wrapper).toMatchSnapshot();
  });
});