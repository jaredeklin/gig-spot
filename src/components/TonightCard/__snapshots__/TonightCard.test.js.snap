// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TonightCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    show={
        Object {
            "artist": "Selina Albright",
            "date": Object {
              "day": "20",
              "dayName": "Fri",
              "month": "Apr",
              "startTime": "20:00",
              "year": "2018",
            },
            "id": 3108932,
            "image": "https://lastfm-img2.akamaized.net/i/u/300x300/2254265bbd13a9aea946244144459709.png",
            "venue": "The Soiled Dove Underground",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="artist"
          className="artist-image"
          src="https://lastfm-img2.akamaized.net/i/u/300x300/2254265bbd13a9aea946244144459709.png"
/>,
        <h4
          className="artist"
>
          Selina Albright
</h4>,
        <h4
          className="venue"
>
          The Soiled Dove Underground
</h4>,
        <h4
          className="date"
>
          Apr 20 20:00
</h4>,
      ],
      "className": "tonight-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "artist",
          "className": "artist-image",
          "src": "https://lastfm-img2.akamaized.net/i/u/300x300/2254265bbd13a9aea946244144459709.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Selina Albright",
          "className": "artist",
        },
        "ref": null,
        "rendered": "Selina Albright",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The Soiled Dove Underground",
          "className": "venue",
        },
        "ref": null,
        "rendered": "The Soiled Dove Underground",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Apr 20 20:00",
          "className": "date",
        },
        "ref": null,
        "rendered": "Apr 20 20:00",
        "type": "h4",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="artist"
            className="artist-image"
            src="https://lastfm-img2.akamaized.net/i/u/300x300/2254265bbd13a9aea946244144459709.png"
/>,
          <h4
            className="artist"
>
            Selina Albright
</h4>,
          <h4
            className="venue"
>
            The Soiled Dove Underground
</h4>,
          <h4
            className="date"
>
            Apr 20 20:00
</h4>,
        ],
        "className": "tonight-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "artist",
            "className": "artist-image",
            "src": "https://lastfm-img2.akamaized.net/i/u/300x300/2254265bbd13a9aea946244144459709.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Selina Albright",
            "className": "artist",
          },
          "ref": null,
          "rendered": "Selina Albright",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The Soiled Dove Underground",
            "className": "venue",
          },
          "ref": null,
          "rendered": "The Soiled Dove Underground",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Apr 20 20:00",
            "className": "date",
          },
          "ref": null,
          "rendered": "Apr 20 20:00",
          "type": "h4",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`TonightCard should match the snapshot 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    show={
        Object {
            "artist": "Selina Albright",
            "date": Object {
              "day": "20",
              "dayName": "Fri",
              "month": "Apr",
              "startTime": "20:00",
              "year": "2018",
            },
            "id": 3108932,
            "image": null,
            "venue": "The Soiled Dove Underground",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="artist"
          className="artist-image"
          src="black-woven.jpg"
/>,
        <h4
          className="artist"
>
          Selina Albright
</h4>,
        <h4
          className="venue"
>
          The Soiled Dove Underground
</h4>,
        <h4
          className="date"
>
          Apr 20 20:00
</h4>,
      ],
      "className": "tonight-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "artist",
          "className": "artist-image",
          "src": "black-woven.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Selina Albright",
          "className": "artist",
        },
        "ref": null,
        "rendered": "Selina Albright",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The Soiled Dove Underground",
          "className": "venue",
        },
        "ref": null,
        "rendered": "The Soiled Dove Underground",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Apr 20 20:00",
          "className": "date",
        },
        "ref": null,
        "rendered": "Apr 20 20:00",
        "type": "h4",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="artist"
            className="artist-image"
            src="black-woven.jpg"
/>,
          <h4
            className="artist"
>
            Selina Albright
</h4>,
          <h4
            className="venue"
>
            The Soiled Dove Underground
</h4>,
          <h4
            className="date"
>
            Apr 20 20:00
</h4>,
        ],
        "className": "tonight-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "artist",
            "className": "artist-image",
            "src": "black-woven.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Selina Albright",
            "className": "artist",
          },
          "ref": null,
          "rendered": "Selina Albright",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The Soiled Dove Underground",
            "className": "venue",
          },
          "ref": null,
          "rendered": "The Soiled Dove Underground",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Apr 20 20:00",
            "className": "date",
          },
          "ref": null,
          "rendered": "Apr 20 20:00",
          "type": "h4",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
